File: easy/shortest_repetition/shortest_repetition.py
import sys

def is_repeatable(sub, original):
    if not (len(original) % len(sub)) and sub * int(len(original) / len(sub)) == original:
        return True
    return False

f = open(sys.argv[1])
lines = f.read().strip().splitlines()

for line in lines:
    match = ''
    shortest = len(line)
    for char in line[0:int(shortest/2)]:
        match += char
        if is_repeatable(match, line):
            shortest = len(match)
            break
    print(shortest)

File: easy/shortest_repetition/shortest_repetition.js
var fs = require('fs');

var isRepetable = function(sub, original) {
    if (!(original.length % sub.length) && Array(original.length/sub.length+1).join(sub) === original) {
        return true;
    } else {
        return false;
    }
}

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== '') {
        var match = '';
        var shortest = line.length;
        line.substring(0, line.length/2).split('').some(function (letter) {
            match += letter;
            if (isRepetable(match, line)) {
                shortest = match.length;
                return true;
            }
        });
        console.log(shortest);
    }
});
